keras/preprocessing/image.py

```diff
@@ -1041,7 +1041,7 @@ def __init__(self, directory, image_data_generator,
             for subdir in sorted(os.listdir(directory)):
                 if os.path.isdir(os.path.join(directory, subdir)):
                     classes.append(subdir)
-        self.num_class = len(classes)
```

Comment:
This change would be a breaking change in theory. However this property is undocumented, and there aren't good use cases for using it, so we will assume it can be considered private.